cpp/src/io/parquet/reader_impl_chunking.cu

```diff
@@ -557,29 +565,43 @@ void reader::impl::compute_input_passes()
                                   ? (rgi.start_row + row_group.num_rows - skip_rows)
                                   : row_group.num_rows;
 
+    // Get the number of leaf-level rows (or number or values) in this row group
+    auto const row_group_leaf_rows =
```

Comment:
I don't think this is going to be sufficient to handle some list edge cases.  It's possible to have a list column with >= 2B rows, but < 2B leaf values.  That's quite a bit harder to handle though - we don't know this until much further down the path. Maybe worth a comment/TODO here.

Reply:
> It's possible to have a list column with >= 2B rows, but < 2B leaf values.

Ah, I see, the `meta_data.num_values` won't count for nulls here 😞. 

>That's quite a bit harder to handle though - we don't know this until much further down the path. Maybe worth a comment/TODO here.

I agree. We would still gracefully error out but I wish there was a better way.

Reply:
>It's possible to have a list column with >= 2B rows, but < 2B leaf values.

I think we can check the number of top-level rows using `row_group.num_rows`, but I agree this would not be error proof for something like: `list<list<int>>`

Reply:
Hi @nvdbaranec, I have added a new check here for top-level rows in 231bd4e for an additional layer of handling as well as a comment to explain the limitations. Also modified the gtest to test the case where total top-level rows >2B but total leaf-values < 2B. 

Please take another look at changes in 231bd4e if you can :)